### PR TITLE
Update docs and add section on templating languages

### PR DESCRIPTION
**Pull request by gabrielpoca**

### CODE REVIEW

File: guides/introduction/getting_started.md [hunk 1]
```diff
@@ -15,7 +17,7 @@ mix still.new mysite
 
 This will create an Elixir project in the folder _mysite_. You'll find an `index.slime` file inside `priv/site`.
 
-Run `mix still.dev` to start the development server. Then open [http://localhost:3000](http://localhost:3000) in your web browser to see the new website.
+Run `mix still.dev` inside `mysite` to start the development server. Then open [http://localhost:3000](http://localhost:3000) in your web browser to see the new website.
```
Comment (frm) [inline guides/introduction/getting_started.md:20]:
```suggestion
Run `cd mysite` and then you can just run `mix still.dev` to start the development server. Then open [http://localhost:3000](http://localhost:3000) in your web browser to see the new website.
```

File: guides/introduction/template_languages.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# Template Languages
+
+The following sections contain more information on the templating languages supported by default and how to add your own.
+
+## Slime
+
+[Slime](https://github.com/slime-lang/slime) is a language that generates HTM. Any file with the extension `.slime` will generate an `.html` file. More information can be found in [Template](https://hexdocs.pm/still/templates.html).
```
Comment (frm) [inline guides/introduction/template_languages.md:7]:
```suggestion
[Slime](https://github.com/slime-lang/slime) is a language that generates HTML. Any file with the extension `.slime` will generate a corresponding `.html` file. More information can be found in [Template](https://hexdocs.pm/still/templates.html).
```

File: guides/introduction/template_languages.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+## EEx
+
+[EEx](https://hexdocs.pm/eex/EEx.html) is not a language, but a tool to embed Elixir code in strings. Files with the `.eex` extension will generate `.html` files (unless you override that configuration). But almost every template run first through EEx, which means that you can use EEx inside Front Matter blocks, Markdown, CSS, JS, almost any text file. More information can be found in [Template](https://hexdocs.pm/still/templates.html).
```
Comment (frm) [inline guides/introduction/template_languages.md:11]:
```suggestion
[EEx](https://hexdocs.pm/eex/EEx.html) is not a language, but a tool to embed Elixir code in strings. Files with the `.eex` extension will generate `.html` files (unless you override that configuration). However, almost every template runs first through EEx, which means that you can use EEx inside Front Matter blocks, Markdown, CSS, JS and almost any other file. More information can be found in [Template](https://hexdocs.pm/still/templates.html).
```

File: guides/introduction/template_languages.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+# Post 1
+
+<%= responsive_image("bg.jpg") %>
+```
+
+More information can be found in [Template](https://hexdocs.pm/still/templates.html).
+
+## CSS
+
+CSS files are regular [templates](https://hexdocs.pm/still/templates.html) where you can embed Elixir:
```
Comment (frm) [inline guides/introduction/template_languages.md:32]: Can we do something like:

```markdown
regular [templates][templates]

[templates]: https://hexdocs.pm/still/templates.html
```

? Does that work?

File: guides/introduction/template_languages.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  defp deps do
+    [
+      ...
+      {:sass, git: "https://github.com/scottdavis/sass.ex", submodules: true},
+    ]
+  end
+
+```
+
+Then, you create a new elixir module somewhere with the following contents:
```
Comment (frm) [inline guides/introduction/template_languages.md:65]:
```suggestion
Then, you create a new Elixir module somewhere with the following contents:
```

File: guides/introduction/templates.md [hunk 6]
```diff
@@ -1,16 +1,16 @@
 # Template
 
-We call template to almost every file in the input folder. These templates
+We call template to almost any file in the input folder. Templates
 are transformed, and combined, to create the website.
 
-## Helpers
+## View Helpers
```
Comment (frm) [inline guides/introduction/templates.md:6]: Should we rename this "Template Helpers" here?

File: guides/introduction/templates.md [hunk 7]
```diff
@@ -43,9 +43,8 @@ subfolders will not work.** For instance:
 
 Image support is still (get it?) rudimentary: it can only generate images of
 different sizes. By default, it relies
-on [ImageMagick][imagemagick], but if you add [`imageflow_ex`][imageflow]
-as a dependency it will use that. The reason we do not use `imageflow_ex`
-by default is that it requires Rust. If you want to use ImageMagick, you
+on [ImageMagick][imagemagick], but if you install [`still_imageflow`](https://github.com/still-ex/still_imageflow)
```
Comment (frm) [inline guides/introduction/templates.md:46]: Can we use the same type of links as elsewhere?

File: guides/introduction/templates.md [hunk 8]
```diff
@@ -67,21 +66,39 @@ A responsive image can be generated like this:
...
 
 ```elixir
 config :still,
   view_helpers: [Your.Module]
 ```
 
-## Layouts
+## Configuration
+
+Most template files allow for a [YAML Front Matter](https://jekyllrb.com/docs/front-matter/) block to change their configuration.
```
Comment (frm) [inline guides/introduction/templates.md:87]: Maybe add a paragraph saying something like:

> If you write add your own template language, you can support this by ensuring the file's preprocessor pipeline includes `Still.Preprocessor.Frontmatter`

?

File: guides/introduction/templates.md [hunk 9]
```diff
@@ -67,21 +66,39 @@ A responsive image can be generated like this:
...
+
+### Permalink
+
+Setting `permalink` allows you to override the output path of a file. For instance, by default `blog/post_1/index.md` would generate `blog/post_1/index.html`, but you can change it to `blog/announcment.html` like this:
```
Comment (frm) [inline guides/introduction/templates.md:91]:
```suggestion
Setting the `permalink` option allows you to override the output path of a file. For instance, by default `blog/post_1/index.md` would generate `blog/post_1/index.html`, but you can change it to `blog/announcment.html` like this:
```

File: guides/introduction/templates.md (hunk 6 above)
Comment (gabrielpoca) [reply guides/introduction/templates.md:6]: I would rather do it altogether in another change.

**Review by frm — APPROVED** [review]

**Pull request merged**